pkg/network/protocols/http/ebpf_gotls.go

```diff
@@ -265,9 +265,21 @@ func (p *GoTLSProgram) Stop() {
 
 func (p *GoTLSProgram) handleProcessStart(pid pid) {
 	exePath := filepath.Join(p.procRoot, strconv.FormatUint(uint64(pid), 10), "exe")
-	binPath, err := os.Readlink(exePath)
+
+	var binPath string
+	var err error
+	// We receive the Exec event, /proc could be slow to update
+	end := time.Now().Add(10 * time.Millisecond)
+	for end.After(time.Now()) {
```

Comment:
I think it is unlikely, but its possible that this `for` loop may never execute and you'll get to a state where  `err` is `nil` and `binPath` is empty

Reply:
yeah true, thanks in really heavy load may the kernel (and gosched) will not schedule us
let's go on the safe side